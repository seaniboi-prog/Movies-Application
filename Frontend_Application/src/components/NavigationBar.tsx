import React from 'react'
import './styles.scss';

type Props = {
    children?: JSX.Element | JSX.Element[];
}

const NavigationBar:React.FC<Props> = ({children}:Props) => {
  return (
    <div className='navigation' data-testid='navigation'>
      <div className='title' data-testid='nav-title'>Movie Gallery</div>
      {children}
    </div>
  )
}

export default NavigationBar
