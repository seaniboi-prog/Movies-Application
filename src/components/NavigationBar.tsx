import React from 'react'
import './styles.scss';

type Props = {
    children?: JSX.Element | JSX.Element[];
}

const NavigationBar:React.FC<Props> = ({children}:Props) => {
  return (
    <div>
      hehe
      {children}
    </div>
  )
}

export default NavigationBar
