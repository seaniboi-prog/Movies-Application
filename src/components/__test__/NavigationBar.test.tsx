// import React from 'react'
// import ReactDOM from 'react-dom'
import NavigationBar from '../NavigationBar'


import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer';

describe('Search Component Tests', () => {

    const title:string = "Movie Gallery";

    const renderComponent = () => {
        render(<NavigationBar />);
    }

    beforeEach(renderComponent);

    afterEach(cleanup);

    it("Renders Search Component correctly", () => {
        const navComponent:HTMLElement = screen.getByTestId('navigation');
        expect(navComponent).toBeInTheDocument();
    });

    it("Renders title div correctly", () => {
        const titleElement:HTMLElement = screen.getByTestId('nav-title');
        expect(titleElement).toBeInTheDocument();
        expect(titleElement).toHaveTextContent(title);
    });

    it("Matches Snapshot", () => {
        const tree = renderer.create(<NavigationBar />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});