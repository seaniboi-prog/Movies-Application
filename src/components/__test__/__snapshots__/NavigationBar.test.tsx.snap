// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component Tests Matches Snapshot 1`] = `
<div
  className="navigation"
  data-testid="navigation"
>
  <div
    className="title"
    data-testid="nav-title"
  >
    Movie Gallery
  </div>
</div>
`;
